src/multiplex/client.rs

```diff
@@ -259,7 +482,7 @@ where
                         tracing::trace!("request sent");
                         drop(guard);
 
-                        this.responses.push_back(Pending {
+                        pending.as_mut().sent(Pending {
```

Comment:
do we want to document the fact that if the user manually removes from the inflight store, this will not be reflected in the `in_flight` counter?

Reply:
Oh, yeah, that seems like a good idea.